repository/src/lib.rs

```diff
@@ -146,9 +146,157 @@ impl<T: RawRepository> DistributedRepository<T> {
     pub async fn fetch(
         &mut self,
         _network_config: &NetworkConfig,
-        _known_peers: &[Peer],
+        known_peers: &[Peer],
     ) -> Result<(), Error> {
-        unimplemented!()
+        for _peer in known_peers {
+            self.raw
+                .add_remote("yoonho".into(), "github".into())
+                .await?;
+
+            // TODO: change yoonho, github into something else with peer
+            // It will be evaluated by "git" method
+        }
+
+        self.raw.fetch_all().await?;
+
+        let branches_after_fetch = self.raw.list_branches().await?;
+
+        // Make block_header_vec to contain BlockHeaders below,
+        // and make block_commit_hash_vec to contain CommitHash
+        let mut block_header_vec: Vec<BlockHeader> = vec![];
+        let mut finalized_branches: Vec<Branch> = vec![];
+
+        // For all incoming branches
```

Comment:
Fix the comment to describe what the whole for loop does.

I think it is pretty obvious  by just looking at L170.